manual/trees/index.md

```diff
@@ -190,13 +199,60 @@ The `TTree::Write()` method is needed to write the ROOT file header.
 
 When writing a {% include ref class="TTree" %} to a ROOT file and if the ROOT file size reaches the value stored in the [TTree::GetMaxTreeSize()](https://root.cern/doc/master/classTTree.html#aca38baf017a203ddb3119a9ab7283cd9){:target="_blank"}, the current ROOT file is closed and a new ROOT file is created. If the original ROOT file is named `myfile.root`, the subsequent ROOT files are named `myfile_1.root`, `myfile_2.root`, etc.
 
-**Autosave**
+_Autosave_
+
+You can flush the tree data to file after the tree has collected a certain amount of data in memory (by default, 300MB).
+If your program crashes, you can recover the ROOT file and the tree entries since the last autosave.
 
-`Autosave` gives you the option to save all branch buffers every n byte. It is recommended to use `Autosave` for large acquisitions. If the acquisition fails to complete, you can recover the ROOT file and all the contents since the last `Autosave`.
+You can adjust the threshold (in bytes or entries) using [TTree::SetAutosave()](https://root.cern/doc/master/classTTree.html#a76259576b0094536ad084cde665c13a8){:target="_blank"}.
 
-- Use the [TTree::SetAutosave()](https://root.cern/doc/master/classTTree.html#a76259576b0094536ad084cde665c13a8){:target="_blank"} method to set the number of bytes between `Autosave`.
+## Reading a tree
 
-You can also use [TTree::SetAutosave()](https://root.cern/doc/master/classTTree.html#a76259576b0094536ad084cde665c13a8){:target="_blank"} in the acquisition loop every n entry.
+> \note **Note**
+>
+> Please use {% include ref class="RDataFrame" namespace="ROOT" %} to read trees, unless you need to do low-level I/O!
+
+To read a tree, you need to associate your variables with the tree's branches, as when writing.
+When loading a tree entry, the tree will set the variables to the branch's value as read from the storage.
+That is done by calling [`TTree::SetBranchAddress()`](https://root.cern/doc/master/classTTree.html#a39b867210e4a77ef44917fd5e7898a1d):
+
+_**Example**_
+
+{% highlight C++ %}
+int variable;
+tree->SetBranchAddress("branchName", &variable);
+for (int iEntry = 0; tree->LoadTree(iEntry) >= 0; ++iEntry) {
+   // Load the data for the given tree entry
+   tree->GetEntry(iEntry);
+
+   // Now, `variable` is set to the value of the branch
+   // "branchName" in tree entry `iEntry`
+   printf("%d\n", variable);
+}
+{% endhighlight %}
+
+In Python you can simply use the branch name as an attribute on the tree:
+
+{% highlight Python %}
+for entry in tree:
+   print(entry.branchName)
```

Comment:
I think iteration over entries is performed this way.  @etejedor can you confirm?

Reply:
He confirms!